providers/monitor.rb

```diff
@@ -6,9 +6,15 @@ def whyrun_supported?
 
 action :add do
   Chef::Log.debug "Adding monitoring for #{new_resource.name}"
-  template "/etc/dd-agent/conf.d/#{new_resource.name}.yaml" do
-    owner 'dd-agent'
-    mode 00600
+  template "#{node['datadog']['config_dir']}/conf.d/#{new_resource.name}.yaml" do
```

Comment:
If we want to be fully cross-platform path-compliant, we should break elements into strings, and use `File.join`:
```
template File.join(node['datadog']['config_dir'], 'conf.d', "#{new_resource.name}.yaml") do
```

This will leverage the OS' native methods, instead of us relying on `/` working across platforms.